gazebo_ros/src/gazebo_ros_factory.cpp

```diff
@@ -344,6 +379,49 @@ void GazeboRosFactoryPrivate::DeleteEntity(
   res->status_message = "Successfully deleted entity [" + req->name + "]";
 }
 
+void GazeboRosFactoryPrivate::SubstPackage(tinyxml2::XMLElement * element)
+{
+  if (!element) {
+    return;
+  }
+
+  for (auto attribute = element->FirstAttribute(); attribute != nullptr;
+    attribute = attribute->Next())
+  {
+    element->SetAttribute(attribute->Name(), ResolvePackage(attribute->Value()).c_str());
```

Comment:
`ResolvePackage` can throw an exception when the package cannot be found or its URL is invalid.
This will stop spawning at the first exception, but I think just providing an error log would be more friendly for the users to find what's missing. Also, it is compatible with the current behavior (the gazebo shows what it can show).